clang/test/SemaHLSL/resource_binding_attr_error_space.hlsl

```diff
@@ -1,5 +1,24 @@
 // RUN: %clang_cc1 -triple dxil-pc-shadermodel6.3-library -x hlsl -o - -fsyntax-only %s -verify
 
+// valid
+cbuffer cbuf {
+    RWBuffer<int> r : register(u0, space0);
+}
+
+cbuffer cbuf2 {
+    struct x {
+        // expected-error@+1 {{'register' attribute only applies to cbuffer/tbuffer and external global variables}}
+        RWBuffer<int> E : register(u2, space3);
+    };
+}

```

Comment:
Presumably we already have a test for this case somewhere else, since the error here is related to the presence of a register attribute at all and not due to register space being there.

Reply:
I thought to add this to show why the `!isDeclaredWithinCOrTBuffer` call is necessary. It prevents the second diagnostic about the space parameter from being emitted. I don't believe there is a case like this that uses the space parameter.

Reply:
Maybe a comment explaining that this test case is explicitly validating that an error isn't emitted would help understand that?